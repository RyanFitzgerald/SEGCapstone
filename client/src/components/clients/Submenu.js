import React from 'react';
import { Link } from 'react-router-dom';

const Submenu = (props) => {
  return (
    <div className="submenu">
      <div className="submenu__actions">
        <ul>
          <li><Link className={(props.activeSubtab === 1) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/clients"><i className="fa fa-list" aria-hidden="true"></i> Client Directory</Link></li>
          {props.checkLevel(JSON.parse(sessionStorage.getItem('user')).role.level, 2) &&
          <li><Link className={(props.activeSubtab === 2) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/clients/add"><i className="fa fa-plus" aria-hidden="true"></i> Add Client</Link></li>
          }
        </ul>
      </div>
    </div>
  );
};

export default Submenu;