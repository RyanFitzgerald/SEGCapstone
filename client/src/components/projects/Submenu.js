import React from 'react';
import { Link } from 'react-router-dom';

const Submenu = (props) => {
  return (
    <div className="submenu">
      <div className="submenu__actions">
        <ul>
          <li><Link className={(props.activeSubtab === 1) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/projects"><i className="fa fa-list" aria-hidden="true"></i> Project Directory</Link></li>
          <li><Link className={(props.activeSubtab === 2) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/projects/add"><i className="fa fa-plus" aria-hidden="true"></i> Add Project</Link></li>
        </ul>
      </div>
    </div>
  );
};

export default Submenu;