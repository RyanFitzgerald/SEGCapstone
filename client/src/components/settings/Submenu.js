import React from 'react';
import { Link } from 'react-router-dom';

const Submenu = (props) => {
  return (
    <div className="submenu">
      <div className="submenu__actions">
        <ul>
          <li><Link className={(props.activeSubtab === 1) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/settings"><i className="fa fa-pencil-square-o" aria-hidden="true"></i> Edit Settings</Link></li>
          <li><Link className={(props.activeSubtab === 2) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/settings/users"><i className="fa fa-list" aria-hidden="true"></i> User Directory</Link></li>
          {props.checkLevel(JSON.parse(sessionStorage.getItem('user')).role.level, 3) &&
          <li><Link className={(props.activeSubtab === 3) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/settings/users/add"><i className="fa fa-plus" aria-hidden="true"></i> Add User</Link></li>
          }
        </ul>
      </div>
    </div>
  );
};

export default Submenu;