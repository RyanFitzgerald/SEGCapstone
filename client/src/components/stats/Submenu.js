import React from 'react';
import { Link } from 'react-router-dom';

const Submenu = (props) => {
  return (
    <div className="submenu">
      <div className="submenu__actions">
        <ul>
          <li><Link className={(props.activeSubtab === 1) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/stats"><i className="fa fa-line-chart" aria-hidden="true"></i> Total Volume</Link></li>
          <li><Link className={(props.activeSubtab === 2) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/stats/type"><i className="fa fa-line-chart" aria-hidden="true"></i> Type Volume</Link></li>
          <li><Link className={(props.activeSubtab === 3) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/stats/salesmen"><i className="fa fa-line-chart" aria-hidden="true"></i> Salesmen Volume</Link></li>
          <li><Link className={(props.activeSubtab === 4) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/stats/referral"><i className="fa fa-line-chart" aria-hidden="true"></i> Referral Volume</Link></li>
        </ul>
      </div>
    </div>
  );
};

export default Submenu;