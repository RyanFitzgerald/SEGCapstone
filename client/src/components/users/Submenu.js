import React from 'react';
import { Link } from 'react-router-dom';

const Submenu = (props) => {
  return (
    <div className="submenu">
      <div className="submenu__actions">
        <ul>
          <li><Link className={(props.activeSubtab === 1) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/users"><i className="fa fa-list" aria-hidden="true"></i> User Directory</Link></li>
          <li><Link className={(props.activeSubtab === 2) ? 'submenu__action submenu__action--active' : 'submenu__action'} to="/users/add"><i className="fa fa-plus" aria-hidden="true"></i> Add User</Link></li>
        </ul>
      </div>
    </div>
  );
};

export default Submenu;