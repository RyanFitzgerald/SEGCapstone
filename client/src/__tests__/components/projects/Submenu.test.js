import React from 'react';
import Submenu from '../../../Components/projects/Submenu';

describe('<Submenu />', () => {
  beforeAll(() => {
    // Add necessary info to user
    const user = {
      role: {
        level: 1
      }
    };
    localStorage.setItem('user', JSON.stringify(user));
  });

  it('should render correctly', () => {   
    // Shallow render it
    const wrapper = shallow(
      <Submenu/>
    );

    // Compare to old snapshot
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    // Check that levels hide elements correctly
    expect(wrapper.find('li')).toHaveLength(1);
  });
});

