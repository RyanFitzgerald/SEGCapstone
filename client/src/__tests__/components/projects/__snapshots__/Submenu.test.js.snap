// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Submenu /> should render correctly 1`] = `
<div
  className="submenu"
>
  <div
    className="submenu__actions"
  >
    <ul>
      <li>
        <Link
          className="submenu__action"
          replace={false}
          to="/projects"
        >
          <i
            aria-hidden="true"
            className="fa fa-list"
          />
           Project Directory
        </Link>
      </li>
    </ul>
  </div>
</div>
`;
